File: src/test/java/com/tuuzed/feedparser/FeedCallbackImpl.java
package com.tuuzed.feedparser;


import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Date;
import java.util.List;


public class FeedCallbackImpl implements FeedCallback {
    private Logger logger = LoggerFactory.getLogger(this.getClass());

    @Override
    public void start() {
        logger.info("");
    }

    @Override
    public void title(@NotNull String title, @Nullable String type) {
        logger.info("title: {}, type: {}", title, type);
    }

    @Override
    public void subtitle(@NotNull String subtitle, @Nullable String type) {
        logger.info("subtitle: {}, type: {}", subtitle, type);
    }

    @Override
    public void link(@NotNull String link, @Nullable String type, @Nullable String title) {
        logger.info("link: {}, type: {}, title: {}", link, type, title);
    }

    @Override
    public void copyright(@NotNull String copyright, @Nullable String type) {
        logger.info("copyright: {}, type: {}", copyright, type);
    }

    @Override
    public void generator(@NotNull String generator, @Nullable String uri, @Nullable String version) {
        logger.info("generator: {}, uri: {}, version: {}", generator, uri, version);
    }

    @Override
    public void end() {
        logger.info("");
    }

    @Override
    public void error(@Nullable Throwable throwable) {

    }

    @Override
    public void fatalError(@Nullable Throwable throwable) {

    }

    @Override
    public void skipDays(@NotNull List<String> skipDays) {
        logger.info("skipDays: {}", skipDays);
    }

    @Override
    public void skipHours(@NotNull List<String> skipHours) {
        logger.info("skipHours: {}", skipHours);
    }

    @Override
    public void itemStart() {
        logger.info("");
    }

    @Override
    public void itemAuthor(@NotNull String author, @Nullable String uri, @Nullable String email) {
        logger.info("author: {}, uri: {}, email: {}", author, uri, email);
    }

    @Override
    public void itemComments(@NotNull String comments) {
        logger.info("comments: {}", comments);
    }

    @Override
    public void itemContent(@NotNull String content, @Nullable String type, @Nullable String language) {
        logger.info("content: {}, type: {}, language: {}", content, type, language);

    }

    @Override
    public void itemContributor(@NotNull String contributor, @Nullable String href, @Nullable String email) {
        logger.info("contributor: {}, href: {}, email: {}", contributor, href, email);
    }

    @Override
    public void itemEnclosure(@Nullable String length, @Nullable String type, @Nullable String url) {
        logger.info("length: {}, type: {}, url: {}", length, type, url);
    }

    @Override
    public void itemExpired(@Nullable Date expired, @NotNull String rawExpired) {
        logger.info("expired: {}, rawExpired: {}", expired, rawExpired);
    }

    @Override
    public void itemId(@NotNull String id) {
        logger.info("id: {}", id);
    }

    @Override
    public void itemLink(@NotNull String link, @Nullable String type, @Nullable String title) {
        logger.info("link: {}, type: {}, title: {}", link, type, title);
    }

    @Override
    public void itemPublished(@Nullable Date published, @NotNull String rawPublished) {
        logger.info("published: {}, rawPublished: {}", published, rawPublished);
    }

    @Override
    public void itemSource(@NotNull String source) {
        logger.info("source: {}", source);
    }

    @Override
    public void itemSummary(@NotNull String summary, @Nullable String type, @Nullable String language) {
        logger.info("summary: {}, type: {}, language: {}", summary, type, language);
    }

    @Override
    public void itemCategory(@NotNull String category, @Nullable String term, @Nullable String scheme) {
        logger.info("category: {}, term: {}, scheme: {}", category, term, scheme);
    }

    @Override
    public void itemTitle(@NotNull String title) {
        logger.info("updated: {}", title);
    }

    @Override
    public void itemUpdated(@Nullable Date updated, @NotNull String rawUpdated) {
        logger.info("updated: {}, rawUpdated: {}", updated, rawUpdated);
    }

    @Override
    public void itemEnd() {
        logger.info("");
    }


}

File: src/test/kotlin/com/tuuzed/feedparser/FeedCallbackImpl.kt
package com.tuuzed.feedparser

import org.slf4j.LoggerFactory
import java.util.*


class FeedCallbackImpl : FeedCallback {
    private val logger = LoggerFactory.getLogger(javaClass)!!
    override fun start() {
        logger.info("")
    }

    override fun title(title: String, type: String?) {
        logger.info("title: $title, type: $type")
    }

    override fun subtitle(subtitle: String, type: String?) {
        logger.info("subtitle: $subtitle, type: $type")
    }

    override fun link(link: String, type: String?, title: String?) {
        logger.info("link: $link, type: $type, title: $title")
    }

    override fun copyright(copyright: String, type: String?) {
        logger.info("copyright: $copyright, type: $type")
    }

    override fun generator(generator: String, uri: String?, version: String?) {
        logger.info("generator: $generator, uri: $uri, version: $version")
    }

    override fun end() {
        logger.info("")
    }

    override fun error(throwable: Throwable?) {
        logger.info(throwable!!.message, throwable)
    }

    override fun fatalError(throwable: Throwable?) {
        logger.info(throwable!!.message, throwable)
    }

    override fun skipDays(skipDays: List<String>) {
        logger.info("skipDays: $skipDays")
    }

    override fun skipHours(skipHours: List<String>) {
        logger.info("skipHours: $skipHours")
    }

    override fun itemStart() {
        logger.info("")
    }

    override fun itemAuthor(author: String, uri: String?, email: String?) {
        logger.info("author: $author, uri: $uri, email: $email")
    }

    override fun itemTitle(title: String) {
        logger.info("title: $title")
    }

    override fun itemId(id: String) {
        logger.info("id: $id")
    }

    override fun itemLink(link: String, type: String?, title: String?) {
        logger.info("link: $link, type: $type, title: $title")
    }

    override fun itemPublished(published: Date?, rawPublished: String) {
        logger.info("published: $published , rawPublished: $rawPublished")
    }

    override fun itemSource(source: String) {
        logger.info("source: $source")
    }

    override fun itemUpdated(updated: Date?, rawUpdated: String) {
        logger.info("updated: $updated, rawUpdated: $rawUpdated")
    }

    override fun itemComments(comments: String) {
        logger.info("comments: $comments")
    }

    override fun itemSummary(summary: String, type: String?, language: String?) {
        logger.info("summary: $summary, type: $type, language: $language")
    }

    override fun itemCategory(category: String, term: String?, scheme: String?) {
        logger.info("category: $category, term: $term, scheme: $scheme")
    }

    override fun itemContent(content: String, type: String?, language: String?) {
        logger.info("type: $type, language: $language, content: $content")
    }

    override fun itemContributor(contributor: String, href: String?, email: String?) {
        logger.info("contributor: $contributor, href: $href, email: $email")
    }

    override fun itemEnclosure(length: String?, type: String?, url: String?) {
        logger.info("length: {$length, type: $length, url: $url")
    }

    override fun itemExpired(expired: Date?, rawExpired: String) {
        logger.info("expired: $expired, rawExpired: $rawExpired")
    }

    override fun itemEnd() {
        logger.info("")
    }
}